services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -1266,7 +1267,8 @@ public BatchFileResult getFileHandleAndUrlBatch(UserInfo userInfo, BatchFileRequ
 							fr.setPreSignedURL(url);
 							FileHandleAssociation association = idToFileHandleAssociation.get(fr.getFileHandleId());
 
-							fileEvents.add(FileEventUtils.buildFileEvent(FileEventType.FILE_DOWNLOAD,userInfo.getId(),association));
+							fileEvents.add(FileEventUtils.buildFileEvent(FileEventType.FILE_DOWNLOAD, userInfo.getId(),
+									association, config.getStack(), config.getStackInstance()));
```

Comment:
It might make sense to store the stack and instance out of the loop and reuse them (Same applies to other parts of the changes). I'm not sure how expensive the calls are though from the config.